tests/integration.rs

```diff
@@ -851,44 +871,39 @@ fn remove_task() -> anyhow::Result<()> {
             &factory_addr,
         )?;
         let manager_addr: Addr = response.metadata.unwrap().contract_addr;
-        let resp = mock.app.borrow_mut().execute_contract(
+        mock.app.borrow_mut().execute_contract(
             Addr::unchecked(AGENT),
             manager_addr,
             &ManagerExecuteMsg::ProxyCall { task_hash: None },
             &[],
         )?;
-        let event = resp
-            .events
-            .into_iter()
-            .find(|ev| {
-                ev.attributes
-                    .iter()
-                    .any(|attr| attr.key == "lifecycle" && attr.value == "task_ended")
-            })
-            .unwrap();
-        let attr = event
-            .attributes
-            .into_iter()
-            .find(|attr| attr.key == "task_hash")
-            .unwrap();
-        attr.value
     };
 
     // Note: not updated
-    let mut active_tasks: Vec<String> = module_contract.active_tasks(None, None)?;
+    let active_tasks: Vec<(Addr, String)> = module_contract.active_tasks(None, None)?;
     assert_eq!(active_tasks.len(), 2);
 
-    active_tasks.retain(|task_hash| task_hash != &removed_task_hash);
-    let not_removed_task_hash = active_tasks.pop().unwrap();
-
+    // Note: tasks sorted by task_hash so we have no idea which one will get executed if they scheduled on a same block, LOL
```

Comment:
Could you elaborate on this? Can this be an issue?

Reply:
Don't worry, since we do not sort explicitly tasks by who came first etc, agent can't be 100% sure which of task he will execute in this block, which is good.
And after task gets executed, one of them gets removed on croncat contracts, but not on the module state, it's done on purpose, so we have not too hacky way to withdraw funds from croncat manager. Modules queries won't check if they exist in current implementation. 
We can make optional argument: `checked` or something, thoughts?

Reply:
Should `checked` be available for both methods? I can see clear use of it on `active_tasks_by_creator`, not sure about regular `active_tasks`
https://github.com/CronCats/abstract-croncat-app/blob/a70527c703048553b3176538eb6f818ee924a7d3/src/msg.rs#L43-L52